lib/src/airship_push.dart

```diff
@@ -98,15 +115,39 @@ class IOSPush {
   Future<bool> isAutoBadgeEnabled() async {
     var isAutoBadgeEnabled = false;
     if (Platform.isIOS) {
-      isAutoBadgeEnabled = await _module.channel.invokeMethod('push#isAutoBadgeEnabled');
+      isAutoBadgeEnabled = await _module.channel.invokeMethod('push#ios#isAutobadgeEnabled');
     }
     return isAutoBadgeEnabled;
   }
 
+  /// Sets the notification options.
+  Future<void> setNotificationOptions(List<NotificationOption> options) async {
+    if (Platform.isIOS) {
+      var parsedOptions = List<NotificationOption>.from(options)
+          .map((option) => option.toString())
+          .toList();
+      return await _module.channel.invokeMethod('push#ios#setNotificationOptions', parsedOptions);
+    } else {
+      return Future.value();
+    }
+  }
+
+  /// Sets the notification options.
+  Future<void> setForegroundPresentationOptions(List<ForegroundPresentationOption> options) async {
```

Comment:
getting errors on this one

Reply:
They need to be converted to the expected string values and not just the enum. I would have an internal method `convertOption` that you switch on the type and return the string. You will need to do that for Features and NotificationOptions as well